### PR TITLE
add cr arbitration punishment logic 

### PR DESCRIPTION
**Pull request by xiaomingloveqinger**

### CODE REVIEW

**Review by peterwillcn — APPROVED** [review]

File: cr/state/committee.go [hunk 1]
```diff
@@ -7,8 +7,10 @@ package state
 
 import (
 	"bytes"
+	"crypto/sha256"
 	"encoding/hex"
 	"errors"
+	"github.com/elastos/Elastos.ELA/crypto"
```
Comment (RainFallsSilent) [inline cr/state/committee.go:13]: change position

File: cmd/common/flags.go [hunk 2]
```diff
@@ -713,6 +713,16 @@ var (
 		Name:  "schnorrstartheight",
 		Usage: "defines the start height to support schnorr transaction",
 	}
+
+	CrArbitrationMonitorHeightFlag = cli.StringFlag{
+		Name:  "crarbitrationmonitorheight",
+		Usage: "defines the start height to monitor cr cross chain transaction",
+	}
+
+	CrArbitrationNotFoundBreachFlag = cli.StringFlag{
```
Comment (RainFallsSilent) [inline cmd/common/flags.go:722]: CrossChainMonitorInterval is better

File: cmd/common/flags.go [hunk 3]
```diff
@@ -713,6 +713,16 @@ var (
 		Name:  "schnorrstartheight",
 		Usage: "defines the start height to support schnorr transaction",
 	}
+
+	CrArbitrationMonitorHeightFlag = cli.StringFlag{
```
Comment (RainFallsSilent) [inline cmd/common/flags.go:717]: CrossChainMonitorStartHeight

File: cmd/common/flags.go (hunk 2 above)
Comment (xiaomingloveqinger) [reply cmd/common/flags.go]: Fixed it

File: cr/state/committee.go (hunk 1 above)
Comment (xiaomingloveqinger) [reply cr/state/committee.go]: Fixed it

File: cr/state/committee.go [hunk 4]
```diff
@@ -1031,6 +1035,146 @@ func (c *Committee) processCRCRealWithdraw(tx interfaces.Transaction,
...
+			}
+			if len(publicKeys) != 0 {
+				break
+			}
+		}
+	}
+	for _, pub := range publicKeys {
+		pubStr := hex.EncodeToString(pub)
+		if !isArbiterEixst(pubStr, c.CurrentSignedWithdrawFromSideChainKeys) {
+			c.CurrentSignedWithdrawFromSideChainKeys = append(c.CurrentSignedWithdrawFromSideChainKeys, pubStr)
```
Comment (RainFallsSilent) [inline cr/state/committee.go:1138]: need to use history

File: cr/state/committee.go [hunk 5]
```diff
@@ -1031,6 +1035,146 @@ func (c *Committee) processCRCRealWithdraw(tx interfaces.Transaction,
...
+					}, func() {
+						tmpMem.MemberState = MemberElected
+						if height >= c.Params.ChangeCommitteeNewCRHeight {
+							c.state.RevertUpdateCRInactivePenalty(tmpMem.Info.CID, height)
+						}
+					})
+				}
+			}
+		}
+		c.CurrentSignedWithdrawFromSideChainKeys = make([]string, 0)
```
Comment (RainFallsSilent) [inline cr/state/committee.go:1164]: need to use history

Comment (xiaomingloveqinger) [reply cr/state/committee.go:1168]: handled

File: cr/state/committee.go (hunk 4 above)
Comment (xiaomingloveqinger) [reply cr/state/committee.go:1142]: handled

**Review by RainFallsSilent — APPROVED** [review]

**Review by Houshoupei84 — APPROVED** [review]

**Pull request merged**